bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/MyRenaultHttpSession.java

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal.renault.api;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.renault.internal.RenaultConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Translation of this python code: https://github.com/hacf-fr/renault-api
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+public class MyRenaultHttpSession {
+
+    private HttpClient httpClient;
+    private Constants constants;
+
+    private Car car;
+    private String kamereonToken;
+    private String kamereonaccountId;
+    private String cookieValue;
+    private String personId;
+    private String gigyaDataCenter;
+    private String jwt;
+
+    private final Logger logger = LoggerFactory.getLogger(MyRenaultHttpSession.class);
+
+    public MyRenaultHttpSession(RenaultConfiguration config) throws Exception {
+
+        car = new Car();
+        this.httpClient = new HttpClient(new SslContextFactory(true));
```

Comment:
No reason other than it was in the class path and worked.  However the "SslContextFactory(true)" hides a sin I forgot I had used to get things working inititally.  I will change this as I need to cert check in on a proper binding.  (sorry) 

What is the best (as in most openHAB friendly) client HTTPClient to use?

Reply:
In your thing handler, initialize your HTTP client like this:
```
this.httpClient = httpClientFactory.getCommonHttpClient();
```
And then pass it to your thing handler.
Most of bindings are doing that, look at the remote openHAB binding as an example.
You create your own HTTP client only when you need a particular HTTP client setup, because of course you must not change the setup of the common client.
In openHAB, the recommended HTTP client is the Jetty client (what is provided by `httpClientFactory.getCommonHttpClient()`. So you will then have no other code to change.

Reply:
Done.  Thanks for the pointer to an implementation to copy.